unified_planning/solvers/solver.py

```diff
@@ -27,6 +30,28 @@ class OptimalityGuarantee(Enum):
     SATISFICING = auto()
     SOLVED_OPTIMALLY = auto()
 
+@dataclass
+class Credits:
+    name: str
+    author: str
+    contact: str
+    website: str
+    short_description: str
+    long_description: str
+
+    def write_credits(self, stream: IO[str], full_credits: bool = False):
+        stream.write(f'You are using {self.name}, developed by {self.author}.\n')
+        if not full_credits:
+            stream.write(self.short_description)
+            stream.write('\n\n')
+        else:
+            stream.write(f'The author can be reached here: {self.contact}\n')
+            stream.write(f'This is the engine website: {self.website}\n')
+            stream.write(self.long_description)
+            stream.write('\n\n')
+
+credits = Credits('Base solver class', 'FBK-team', 'aiplan4eu@fbk.eu', 'https://github.com/aiplan4eu/unified-planning', 'Solver base class', 'Solver base class')
```

Comment:
Why creating this instance here? I think it would be far better to have a staticmethod in the `Solver` class that returns a `Credits` instance (that is then printed from outside)